Translation updates

wbobeirne opened the pull request:
Just a quick PR to close out #76:

* Updates all the translation files
* Moves them into a `lang/` subdirectory
* Uses the `react-markdown` module instead of hand-writing regexs for our own markdown implementation
    * v3 uses the `marked` library, so it already was using a library for this, and I didn't notice any significant performance implications with this change
    * I also removed the `markdown.spec.js` tests as a result of this

I noticed one problem, which I didn't address but instead opened #121 for, is that not all translations re-render upon changing the language. This is because not all components have the language as a prop, so when that changes the component stays the same, and doesn't re-render. Let's keep discussion of that problem on the relevant issue though, it doesn't have any bearing on this update.

dternyak commented:
@wbobeirne Nice!

It looks like you have some flow errors however.

```
16.84s$ npm run flow
> MyEtherWallet@0.0.1 flow /home/travis/build/MyEtherWallet/MyEtherWallet
> flow
Launching Flow server for /home/travis/build/MyEtherWallet/MyEtherWallet
Spawned flow server (pid=6364)
Logs will go to /tmp/flow/zShomezStraviszSbuildzSMyEtherWalletzSMyEtherWallet.log
common/containers/Tabs/Swap/components/ReceivingAddress.jsx:86
                 v------------
 86:             <SimpleButton
 87:               text={translate('SWAP_start_CTA')}
 88:               onClick={this.onClickPartTwoComplete}
...:
 91:             />
                 -^ props of React element `SimpleButton`. This type is incompatible with
 31:   props: Props;
              ^^^^^ object type. See: common/components/ui/SimpleButton.jsx:31
  Property `text` is incompatible:
     87:               text={translate('SWAP_start_CTA')}
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^ React$Element. This type is incompatible with
     22:   text: string,
                 ^^^^^^ string. See: common/components/ui/SimpleButton.jsx:22
common/sagas/wallet.js:61
 61:     yield put(showNotification('danger', translate('INVALID_PKEY')));
                                              ^^^^^^^^^^^^^^^^^^^^^^^^^ React$Element. This type is incompatible with the expected param type of
 21:   msg: string,
            ^^^^^^ string. See: common/actions/notifications.js:21
Found 2 errors
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! MyEtherWallet@0.0.1 flow: `flow`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the MyEtherWallet@0.0.1 flow script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! A complete log of this run can be found in:
npm ERR!     /home/travis/.npm/_logs/2017-08-16T17_37_53_576Z-debug.log
The command "npm run flow" exited with 2.
```

crptm commented:
I think we used plaintext feature in some places, we have to double check that every translation point accepts react components

wbobeirne commented:
I fixed the flow types so that showNotification can accept react elements as well as strings (Or any other renderable.) I changed it from using dangerouslysetinnerhtml to just rendering the message, and fixed the couple of spots that were sending strings of HTML instead of React elements. I don't think this should cause any trouble.

wbobeirne commented:
@dternyak that's not a regression from this PR, it's a problem that exists in `develop` already due to React not re-rendering. I didn't change how components update with this, so we could solve #121 independently of this change.

But I'll go ahead and try to roll it in with this PR.

wbobeirne commented:
Alright, just pushed `components/Translate.jsx` which is a connected component that pulls the language from redux store. Replaced `translate` to return that component, and added `translateRaw` which will allow us to grab raw text instead, if that's needed anywhere.

dternyak commented:
Looks like you're right -- this is a problem that already exists in develop.

There are a few flow errors, so I will fix them and then merge.

dternyak commented:
@wbobeirne: @crptm Was correct that there are several places where we expect plain text (e.g. placeholders).

I've gone with the approach of having a `translate` function that gets the current state from the store and returns the string instead of the connected component.  

```
export function translateRaw(key: string, lang: string) {
  return (
    (repository[lang] && repository[lang][key]) ||
    repository[fallbackLanguage][key] ||
    key
  );
}

export function translateRawFromState(key: string) {
  const state = store.getState();
  const lang = getLanguageSelection(state);
  return translateRaw(key, lang)
}
```

We can revert the commit if there is disagreement about this approach.

dternyak commented on common/components/ui/Modal.jsx:
```diff
@@ -18,8 +18,7 @@ type Props = {
     disabled?: boolean,
-    // $FlowFixMe - Why the fuck doesn't this like onClick?
-    onClick: () => void
+    onClick?: () => void
```
@wbobeirne - I think you previously had asked for advice here, there were some cases where the onclick prop was not even specified (different than specified with a null value).

In this case you need to mark it as a an [optional property](https://flow.org/en/docs/types/objects/#toc-optional-object-type-properties)

wbobeirne replied:
Definitely aware, and not something I wanted those buttons to have (A button in a modal should always do something.) I was passing onClick, but it was still complaining, so I haven't quite figured that one out yet.

wbobeirne commented:
I'm satisfied with that approach if refreshing is off the table, but I do think that would reduce a considerable amount of complexity. Thanks for pitching in on this.

bfillmer commented:
@dternyak Fixed flow errors and a few missing translation pieces. Probably want to put this through it's paces but should be good to go.
